packages/rest/test/unit/rest.server/rest.server.unit.ts

```diff
@@ -86,6 +86,40 @@ describe('RestServer', () => {
       expect(server.getSync(RestBindings.PORT)).to.equal(4000);
       expect(server.getSync(RestBindings.HOST)).to.equal('my-host');
     });
+
+    it('honors basePath in config', async () => {
+      const app = new Application({
+        rest: {port: 0, basePath: '/api'},
+      });
+      app.component(RestComponent);
+      const server = await app.getServer(RestServer);
+      expect(server.getSync(RestBindings.BASE_PATH)).to.equal('/api');
+    });
+
+    it('honors basePath via api', async () => {
+      const app = new Application({
+        rest: {port: 0},
+      });
+      app.component(RestComponent);
+      const server = await app.getServer(RestServer);
+      server.basePath('/api');
+      expect(server.getSync(RestBindings.BASE_PATH)).to.equal('/api');
+    });
+
+    it('rejects basePath if request handler is created', async () => {
+      const app = new Application({
+        rest: {port: 0},
+      });
+      app.component(RestComponent);
+      const server = await app.getServer(RestServer);
+      expect(() => {
+        if (server.requestHandler) {
```

Comment:
If this condition happens to be false, then `server.basePath` is not invoked at all and the test fails with a confusing error.

Please convert this `if` into an explicit assertion. For example:

```ts
expect.assert(!!server.requestHandler, 'requestHandler should have been set up by now');
expect(() => server.basePath('/api')).to.throw(/*...*/);
```